File: 백준/Gold/1107. 리모컨/리모컨.kt
fun main() {
    val N = readLine()!!.toInt()
    val M = readLine()!!.toInt()
    val broken: List<String> = if (M != 0) {
        readLine()!!.split(" ").map { it }
    } else {
        emptyList()
    }

    var answer = Math.abs(100 - N)

    for (num in 0..1_000_000) {
        var passed = true
        for (i in num.toString()) {
            if (broken.contains(i.toString())) {
                passed = false
                break
            }
        }
        if (passed) answer = Math.min(answer, num.toString().length + Math.abs(num - N))
    }
    println(answer)

}

File: 백준/Gold/1107. 리모컨/리모컨.py
import sys
import math

target = int(sys.stdin.readline())
ans = abs(100-target)
m = int(sys.stdin.readline())
if m:
    numset = set(sys.stdin.readline().split())
else:
    numset = set()

for num in range(10000001):
    for N in str(num):
        if N in numset:
            #print("걸림")
            break
    else:
        ans = min(ans,len(str(num)) + abs(num-target))
print(ans)